src/repo.rs

```diff
@@ -2346,6 +2347,23 @@ impl Repository {
         }
         Ok(ignored == 1)
     }
+
+    /// Perform a cherrypick
+    pub fn cherrypick(
+        &self,
+        commit: &Commit<'_>,
+        options: Option<&mut CherrypickOptions<'_>>,
+    ) -> Result<(), Error> {
+        unsafe {
+            let raw_opts = match options {
+                Some(opts) => &opts.raw(),
```

Comment:
I think this needs to be moved into a temporary because otherwise this creates a raw pointer which, when used, is a use-after-free

Reply:
Indeed, that was a mistake.

I'm not entirely familiar with Rust, but I believe I fixed it. I suppose Rust stores an Option<git_cherrypick_options> as a struct + bool, so pattern matching + take the ref of the Some value should be ok as long as it is used in the same scope as the Option.